src/hippopt/robot_planning/expressions/kinematics.py

```diff
@@ -0,0 +1,251 @@
+    base_quaternion_derivative = cs.MX.sym(base_quaternion_xyzw_derivative_name, 4)
+    joint_velocities = cs.MX.sym(joint_velocities_name, kindyn_object.NDoF)
+
+    cmm_function = kindyn_object.centroidal_momentum_matrix_fun()
+
+    base_pose = liecasadi.SE3.from_position_quaternion(
+        base_position, base_quaternion
+    ).as_matrix()  # The quaternion is supposed normalized
+
+    base_angular_velocity_fun = (
+        quaternion_xyzw_velocity_to_right_trivialized_angular_velocity(
+            quaternion_xyzw_name=base_quaternion_xyzw_name,
+            quaternion_xyzw_velocity_name=base_quaternion_xyzw_derivative_name,
+            options=options,
+        )
+    )
+    base_angular_velocity = base_angular_velocity_fun(
+        **{
+            base_quaternion_xyzw_name: base_quaternion,
+            base_quaternion_xyzw_derivative_name: base_quaternion_derivative,
+        }
+    )["right_trivialized_angular_velocity"]
+
+    robot_velocity = cs.vertcat(
+        base_position_derivative, base_angular_velocity, joint_velocities
+    )
+
+    momentum = cmm_function(base_pose, joint_positions) @ robot_velocity
+
+    return cs.Function(
+        "centroidal_momentum_from_kinematics",
+        [
+            base_position,
+            base_quaternion,
+            joint_positions,
+            base_position_derivative,
+            base_quaternion_derivative,
+            joint_velocities,
+        ],
+        [momentum],
+        [
+            base_position_name,
+            base_quaternion_xyzw_name,
+            joint_positions_name,
+            base_position_derivative_name,
+            base_quaternion_xyzw_derivative_name,
+            joint_velocities_name,
+        ],
+        ["h_g"],
+        options,
+    )
+
+
+def center_of_mass_position_from_kinematics(
+    kindyn_object: KinDynComputations,
+    base_position_name: str = "base_position",
+    base_quaternion_xyzw_name: str = "base_quaternion",
+    joint_positions_name: str = "joint_positions",
+    options: dict = None,
+    **_
+) -> cs.Function:
+    options = {} if options is None else options
+
+    base_position = cs.MX.sym(base_position_name, 3)
+    base_quaternion = cs.MX.sym(base_quaternion_xyzw_name, 4)
+    joint_positions = cs.MX.sym(joint_positions_name, kindyn_object.NDoF)
+
+    com_function = kindyn_object.CoM_position_fun()
+
+    base_pose = liecasadi.SE3.from_position_quaternion(
+        base_position, base_quaternion
+    ).as_matrix()  # The quaternion is supposed normalized
+
+    com_position = com_function(base_pose, joint_positions)
+
+    return cs.Function(
+        "center_of_mass_position_from_kinematics",
+        [
+            base_position,
+            base_quaternion,
+            joint_positions,
+        ],
+        [com_position],
+        [
+            base_position_name,
+            base_quaternion_xyzw_name,
+            joint_positions_name,
+        ],
+        ["com_position"],
+        options,
+    )
```

Comment:
Suggestion: I always suggest to return `.expand()` functions since casadi complains if the developer made mistakes in the input/output symbols. Not sure though what could be the implications on the graph. Keeping the output as MX instead SX (and just expand as late as possible) would result to faster graph evaluation?

Reply:
Since I output the function, I would leave this choice to the end user.